File: packages/posthog/ios/Plugin/Classes/Options/SetupOptions.swift
import Foundation

@objc public class SetupOptions: NSObject {
    private var apiKey: String
    private var host: String
    private var enableSessionReplay: Bool
    private var sessionReplayConfig: SessionReplayOptions?
    private var config: [String: Any]?

    init(apiKey: String, host: String, enableSessionReplay: Bool, sessionReplayConfig: [String: Any]?, config: [String: Any]? = nil) {
        self.apiKey = apiKey
        self.host = host
        self.enableSessionReplay = enableSessionReplay
        self.config = config

        if let replayConfig = sessionReplayConfig {
            self.sessionReplayConfig = SessionReplayOptions(
                screenshotMode: replayConfig["screenshotMode"] as? Bool ?? false,
                maskAllTextInputs: replayConfig["maskAllTextInputs"] as? Bool ?? true,
                maskAllImages: replayConfig["maskAllImages"] as? Bool ?? true,
                maskAllSandboxedViews: replayConfig["maskAllSandboxedViews"] as? Bool ?? true,
                captureNetworkTelemetry: replayConfig["captureNetworkTelemetry"] as? Bool ?? true,
                debouncerDelay: replayConfig["debouncerDelay"] as? Double ?? 1.0
            )
        }
    }

    func getApiKey() -> String {
        return apiKey
    }

    func getHost() -> String {
        return host
    }

    func getEnableSessionReplay() -> Bool {
        return enableSessionReplay
    }

    func getSessionReplayConfig() -> SessionReplayOptions? {
        return sessionReplayConfig
    }

    func getConfig() -> [String: Any]? {
        return config
    }
}

File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/SetupOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java.util.Map;

public class SetupOptions {

    @NonNull
    private String apiKey;

    @NonNull
    private String host;

    @Nullable
    private Boolean enableSessionReplay;

    @Nullable
    private SessionReplayOptions sessionReplayConfig;

    @Nullable
    private Map<String, Object> config;

    public SetupOptions(@NonNull String apiKey, @NonNull String host) {
        this.apiKey = apiKey;
        this.host = host;
        this.config = null;
    }

    public SetupOptions(@NonNull String apiKey, @NonNull String host, @Nullable Map<String, Object> config) {
        this.apiKey = apiKey;
        this.host = host;
        this.config = config;
    }

    @NonNull
    public String getApiKey() {
        return apiKey;
    }

    @NonNull
    public String getHost() {
        return host;
    }

    @Nullable
    public boolean getEnableSessionReplay() {
        return enableSessionReplay;
    }

    public void setEnableSessionReplay(boolean enableSessionReplay) {
        this.enableSessionReplay = enableSessionReplay;
    }

    @Nullable
    public SessionReplayOptions getSessionReplayConfig() {
        return sessionReplayConfig;
    }

    public void setSessionReplayConfig(@Nullable SessionReplayOptions sessionReplayConfig) {
        this.sessionReplayConfig = sessionReplayConfig;
    }

    @Nullable
    public Map<String, Object> getConfig() {
        return config;
    }
}
